core/src/main/scala/org/http4s/StaticFile.scala

```diff
@@ -126,5 +126,11 @@ object StaticFile {
     readAll[Task](f.toPath, DefaultBufferSize)
   }
 
+  private def nameToContentType(name: String): Option[`Content-Type`] =
+    name.lastIndexOf('.') match {
+      case -1 => None
+      case i => MediaType.forExtension(name.substring(i + 1)).map(`Content-Type`(_))
```

Comment:
What happens if name is just `.`?  I can't remember whether substring out of bounds returns an empty string or throws.